server/gateway.go

```diff
@@ -1072,7 +1072,26 @@ func (c *client) processGatewayInfo(info *Info) {
 		// connect events to switch those accounts into interest only mode.
 		s.mu.Lock()
 		s.ensureGWsInterestOnlyForLeafNodes()
+		js := s.js
 		s.mu.Unlock()
+
+		// Switch JetStream accounts to interest-only mode.
+		if js != nil {
```

Comment:
I should do that only if s.gateway.enabled is true, let me fix that.

Reply:
Well, clearly it is enabled if we process an inbound GW, so no need to change anything :-)